agent/agent_test.go

```diff
@@ -771,13 +771,15 @@ tasks:
   - name: this_is_a_task_name
     commands:
       - command: shell.exec
+        failure_metadata_tags: ["failure_tag0"]
         params:
           script: exit 0
 
 post_error_fails_task: true
 post_timeout_secs: 1
 post:
   - command: shell.exec
+    failure_metadata_tags: ["failure_tag1"]
```

Comment:
nit: can we check that this also works when multiple tags are set

Reply:
Multiple tags should already covered by [this test](https://github.com/evergreen-ci/evergreen/blob/f7345914f8c5d418c77c496f87f01592922c2fb4/agent/agent_test.go#L1715), which is checking that it correctly gets the set union of all the failure metadata tags,